tests/test_cli.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+from argparse import ArgumentParser
+
+from .common import test_result_collector as trc
+
+from model_analyzer.config.input.config_command_profile import ConfigCommandProfile
+from model_analyzer.config.input.config_command_analyze import ConfigCommandAnalyze
+from model_analyzer.config.input.config_command_report import ConfigCommandReport
+from model_analyzer.cli.cli import CLI
+
+
+class ArgumentParserSubclass(ArgumentParser):
+
+    def print_help(self, file=None):
+        super().print_help(file)
+        self._did_print_help = True
+
+    _did_print_help = False
+
+
+class CLISubclass(CLI):
+
+    def __init__(self):
+        self._parser = ArgumentParserSubclass()
+        self._add_global_options()
+        self._subparsers = self._parser.add_subparsers(
+            help='Subcommands under Model Analyzer', dest='subcommand')
+
+        # Store subcommands, and their configs
+        self._subcommand_configs = {}
+
+
+class TestCLI(trc.TestResultCollector):
+
+    def test_help_message_no_args(self):
```

Comment:
Please comment the test, as well as the above classes

Also, is "subclass" the appropriate term?  Didn't we use Test* in DLProf:   PytorchIterationTracker vs TestPytorchIterationTracker.  Does that not work because this testing framework looks for anything that starts with Test* to run?

Reply:
added comments 👍

to use python [unittest](https://docs.python.org/3/library/unittest.html#basic-example), you basically create a subclass of `unittest.TestCase`. that subclass has to have a name, of course, and in this case, the file I added is supposed to test the CLI, so the files name is `test_cli.py` and the subclass of `unittest.TestCase` has the same name (TestCLI), to match the pattern of other unit tests (e.g. `test_run_search.py` has `TestRunSearch` as the subclass of `unittest.TestCase`)

that left me in a conundrum, because we used to do the thing where if we were trying to write a test for `SomeSpecificClass`, but we needed to change functionality of a particular method to test it, we'd make a subclass of `SomeSpecificClass` called `TestSomeSpecificClass`, but that clearly can't work with the above naming pattern

hence i used `SomeSpecificClassSubclass`

any suggestions on alternative ways of handling this naming pattern conflict?

Reply:
Maybe you can mock with google mock (Tim mentioned this in another comment) so that `ArgumentParserSubclass` can just be a mocked internal member?

Reply:
switched to using [unittest.mock](https://docs.python.org/3/library/unittest.mock.html) 👍